spec/requests/orders_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+RSpec.describe "Orders", type: :request do
+  # describe "GET /index" do
+  #   pending "add some examples (or delete) #{__FILE__}"
+  # end
+  describe "get orders_path" do
+    it "renders the index view" do
+      FactoryBot.create_list(:order, 10)
+      get orders_path
+      expect(response).to render_template(:index)
+    end
+  end
+  describe "get order_path" do
+    it "renders the :show template" do
+      order = FactoryBot.create(:order)
+      get order_path(id: order.id)
+      expect(response).to render_template(:show)
+    end
+    it "redirects to the index path if the order id is invalid" do
+      get order_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to orders_path
+    end
+  end
+  describe "get new_order_path" do
+    it "renders the :new template" do
+      get new_order_path
+      expect(response).to render_template(:new)
+    end
+  end
+  describe "get edit_order_path" do
+    it "renders the :edit template"  do
+      order = FactoryBot.create(:order)
+      get edit_order_path(id: order.id)
+      expect(response).to render_template(:edit)
+    end
+  end
+  describe "post orders_path with valid data" do
+    it "saves a new entry and redirects to the show path for the entry" do
+      customer = FactoryBot.create(:customer)
+      order_attributes = FactoryBot.attributes_for(:order)
+      order_attributes[:customer_id]=customer.id
```

Comment:
Just shorthand for what you did:

```suggestion
      order_attributes = FactoryBot.attributes_for(:order, customer_id: customer.id)
```